flow-typed/src/ChorusClient.js

```diff
@@ -0,0 +1,12 @@
+interface ChorusClient {
+	open(url: string): Promise<Event>,
+	close(): void,
+	connect(): void,
+	publishStep(
+		pattern: string,
+		callback: (...args: Array<string>) => number | string | void,
```

Comment:
So at present the callback to process a Step gets and array of String (which the matches from any regular expression capture groups), and returns either a number, a string or void. 

I think it's definitely right for Chrous to send you the matches as a List/Array of Strings, and I think its also probably right for you to pass those on just so to the callback function. One aspect of the Java approach to this, since we're dealing in Java with Types which can be accessed via reflection at runtime, is that Chorus has some code will try to do some automatic 'Type Coersion' to try to covert those Strings automatically into the correct types that the callback function (or Method in Java) requires as arguments. I guess all Type information is lost during the transpilation process, so this wouldn't work for JS? That really doesn't matter, it's a very low importance thing, but just an interesting discussion.